Refactoring APIs

itTkm opened the pull request:
Hi @lianginx .
I primarily refactored the implementation of server-side APIs to improve future maintainability and enhance extensibility.

The points for refactoring are as follows:
- Separated the internal API endpoints based on their purposes. (And also following the specifications of OpenAI's endpoint.)
- The internal processing parameters are sent in the request header, not in the request body, to support the GET method.
- The server-side outputs console logs.

Testing
- [x] OpenAI API, configuration is provided on the application.
- [x] Azure OpenAI Service, configuration is provided on the application.
- [x] OpenAI API, configuration is provided with environment variables.
- [x] Azure OpenAI Service, configuration is provided with environment variables.

lianginx commented:
When I used the wrong API Key, there was no correct prompt returned. Instead, a 404 error occurred. When I changed to the correct API Key, it returned to normal.

I searched for the issue https://github.com/nuxt/nuxt/issues/15293#issuecomment-1397375873, it seems to be caused by this reason.

![image](https://github.com/lianginx/chatgpt-nuxt/assets/21002250/85939146-fba7-4592-ab54-51d05d0544e5)

![image](https://github.com/lianginx/chatgpt-nuxt/assets/21002250/a4216850-8c34-4ced-bfa3-008af9865184)

itTkm commented:
@lianginx 
Thank you for your report. I have faced same issue.

I have fixed this issue with a1acfa4fa5ae8325b1d1eebca593304f2a4bb8c7 as bellow:

### OpenAI API
<img width="1003" alt="image" src="https://github.com/lianginx/chatgpt-nuxt/assets/61522301/d1151239-6171-4d6a-a9a6-62af55bb5508">

In that case, the API does not provide an error response message. 
Therefore, I have implemented displaying the status code and status text.

```json
{
    "error": {
        "message": "",
        "type": "invalid_request_error",
        "param": null,
        "code": "invalid_api_key"
    }
}
```

### Azure OpenAI Service
<img width="1000" alt="image" src="https://github.com/lianginx/chatgpt-nuxt/assets/61522301/93737c11-2d61-401a-b848-302228969a6b">

```json
{
    "statusCode": 401,
    "message": "Unauthorized. Access token is missing, invalid, audience is incorrect (https://cognitiveservices.azure.com), or have expired."
}
```
